### DISCUSSION TITLE
Discussion: shift events recurring on the 29/30/31 of every month to the last day of the month?

### DISCUSSION THREAD

**Discussion by derekantrican** [derekantrican/GAS-ICS-Sync#310] [Ideas]
If you use Outlook and set the following recurrence:

![image](https://user-images.githubusercontent.com/1558019/215161221-49fd3ba0-5d22-4d33-be7f-58f12d96a5b8.png)

and click "OK" to save, you will get the following message:

![image](https://user-images.githubusercontent.com/1558019/215161278-6e0e483a-204b-417d-aec1-e6919f92ef46.png)

This will happen if you set an event to repeat on the 29th, 30th, or 31st of every month. It seems to be a design choice that Outlook has made as Google Calendar _does not do the same thing_.

Should we...

1. Maintain the original purpose of this script as "Google Calendar ICS syncing _but faster_"
2. Implement a setting something like `shiftEndOfMonthEvents` to allow users to follow the Outlook convention?

@jonas0b1011001 as FYI

**Comment by jonas0b1011001** [comment]
The thing is, that we are currently simply passing the recurrence rule _as is_ to google and let them do whatever is needed. If we change that we would probably have to create **all** recurrence instances manually.

**Reply by fennelpollen** [reply]
I'm not sure what's the smartest solution, and it's a problem if Google changes the rule interpretation in the future. However, as a user, I'd just like to see exact same behavior as it is in Outlook. Since this is from Outlook to Google (not the other way around). Is there a reason why we can't just implement the 2nd solution (shiftEndOfMonthEvents)? Why do we have to create all recurrence instances manually then?

**Reply by fennelpollen** [reply]
There is another problem arising from this. I just tested and this happens: I create an event on 31st, recurring monthly. On a month with less than 31 days, in Outlook, I move the event from 30th to 6th. Then the event on 6th appears on Google Calendar. But when deleting that 6th day event in Outlook, it doesn't disappear from Google Calendar.

Would be great if this just worked without problems from user's perspective. Anyway, thank you for your time.

**Reply by jonas0b1011001** [reply]
This is with the script syncing or google's subscription feature?
